src/create-template-orders.js

```diff
@@ -18,32 +17,50 @@ const LAST_START_TIMESTAMP_CUSTOM_OBJECT_KEY =
 // to address all the write inconsistencies when writing to database
 const INCONSISTENCY_MS = 3 * 60 * 1000
 
-const stats = {
-  processedCheckoutOrders: 0,
-  createdTemplateOrders: 0,
-  failedCheckoutOrders: 0,
-  duplicatedTemplateOrderCreation: 0,
-}
+async function createTemplateOrders({
+  apiRoot: _apiRoot,
+  ctpClient: _ctpClient,
+  logger: _logger,
+  startDate,
+}) {
+  stats = {
+    processedCheckoutOrders: 0,
+    createdTemplateOrders: 0,
+    failedCheckoutOrders: 0,
+    duplicatedTemplateOrderCreation: 0,
+  }
+
+  try {
+    apiRoot = _apiRoot
+    ctpClient = _ctpClient
+    logger = _logger
 
-async function createTemplateOrders(startDate) {
-  const uri = await _buildFetchCheckoutOrdersUri()
+    const uri = await _buildFetchCheckoutOrdersUri()
 
-  await ctpClient.fetchBatches(uri, async (orders) => {
-    stats.processedCheckoutOrders += orders.length
-    await pMap(orders, _processCheckoutOrder, { concurrency: 3 })
-  })
+    await ctpClient.fetchBatches(uri, async (orders) => {
+      stats.processedCheckoutOrders += orders.length
```

Comment:
I feel that increment should be done one by one on _processCheckoutOrder 🤔 instead of increment with total number

Reply:
https://github.com/commercetools/commercetools-subscriptions/pull/6/commits/8cf1a3593746ada5c809851bceb3f2d80fc26450#diff-33e44438e7461ee9bc6d2f6720f5f57e1aea2806100eb8ca03d98eb8b2124fc5